Marlin/src/feature/tmc_util.cpp

```diff
@@ -961,12 +1024,20 @@
     st.en_pwm_mode(restore_stealth);
     st.diag1_stall(false);
   }
-  bool tmc_enable_stallguard(TMC2660Stepper) {
+  bool tmc_enable_stallguard(TMC2660Stepper &st) {
     // TODO
     return false;
   }
   void tmc_disable_stallguard(TMC2660Stepper, const bool) {};
 
+  // TCOOLTHRS ≥ TSTEP > TPWMTHRS  Stall output signal on pin DIAG is enabled
+  bool tmc_enable_stallguard(TMC2209Stepper &st) {
+    st.TCOOLTHRS(0xFFFFF);
+    return true;
+  }
+  void tmc_disable_stallguard(TMC2209Stepper &st, const bool restore_stealth) {
+    st.TCOOLTHRS(0);
+  }
```

Comment:
@teemuatlut — The `TMC2209Stepper` class doesn't have a `TCOOLTHRS` method. So I assume these two functions are incorrect.

Reply:
Actually it should have the method but I've just missed it. Adding it now but it'll be included in the next release.

Sorry this is taking a while for me to start reviewing but the current issues have taken priority.

Reply:
No problem. We've got plenty to keep us occupied in the meantime!